src/main/java/org/openrewrite/java/testing/junit5/TemporaryFolderToTempDir.java

```diff
@@ -93,187 +100,181 @@ protected TemporaryFolderToTempDirVisitor getVisitor() {
-                        if (fieldName.equals(receiver.getSimpleName())) {
-                            List<Expression> args = m.getArguments();
-                            // handle TemporaryFolder.newFile() and TemporaryFolder.newFile(String)
-                            switch (m.getName().getSimpleName()) {
-                                case "newFile":
-                                    if (args.size() == 1 && args.get(0) instanceof J.Empty) {
-                                        m = m.withTemplate(
-                                                template("File.createTempFile(\"junit\", null, " + fieldName + ");").build(),
-                                                m.getCoordinates().replace()
-                                        );
-                                    } else {
-                                        doAfterVisit(new AddNewFileMethod(tempDirFieldVar.getName(), method));
-                                    }
-                                    break;
-                                case "getRoot":
-                                    return receiver.withPrefix(m.getPrefix()).withType(tempDirFieldVar.getType());
-                                case "newFolder":
-                                    if (args.size() == 1 && args.get(0) instanceof J.Empty) {
-                                        m = m.withTemplate(template("Files.createTempDirectory(#{any(java.io.File)}.toPath(), \"junit\").toFile();").imports("java.nio.file.Files", "java.io.File")
-                                                .build(), m.getCoordinates().replace(), tempDirFieldVar.getName());
-                                        maybeAddImport("java.nio.file.Files");
-                                    } else {
-                                        doAfterVisit(new AddNewFolderMethod(tempDirFieldVar.getName(), method));
-                                    }
-                                    break;
-                            }
-                        }
-                    }
+        @Override
+        public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, executionContext);
+            String declaringType = mi.getType() != null ? mi.getType().getDeclaringType().getFullyQualifiedName() : null;
+            if ("org.junit.rules.TemporaryFolder".equals(declaringType) && mi.getSelect() != null) {
+                switch (mi.getSimpleName()) {
+                    case "newFile":
+                        return convertToNewFile(mi);
+                    case "newFolder":
+                        doAfterVisit(new AddNewFolderMethod(mi));
+                        break;
+                    case "create":
+                        return null;
+                    default:
+                        return mi.getSelect().withPrefix(Space.format(" "));
```

Comment:
why are we changing the select's prefix? when would we hit the default here ?

Reply:
Thanks, @tkvangorder.  The `select` has an empty prefix which would result in `File root =tempDir;`.
Setting the prefix to a single space is fragile; it should actually be the prefix from the method invocation.
`return mi.getSelect().withPrefix(mi.getPrefix());`